### DISCUSSION TITLE
change point size of add_treetops3d

### DISCUSSION THREAD

**Discussion by lindfla** [r-lidar/lidR#496] [Q&A]
Hi Jean-Romain,

I am working with your amazing lidR package these days and love it.

I want to plot my field data (x, y, z values  that I convert to a sp data frame) above the lidar data.
For that I am using the `plot_3d` and the `add_treetops3D` addition.

I have changed your example code, so that the added points are not treetops but my actual measured tree heights and locations.

This is the code:

```r
plotdata
#> class       : SpatialPointsDataFrame 
#> features    : 51 
#> extent      : 593576.7, 593586.8, 6814138, 6814148  (xmin, xmax, ymin, ymax)
#> crs         : +proj=utm +zone=10 +datum=WGS84 +units=m +no_defs 
#> variables   : 2
#> names       :   ID,    Z 
#> min values  :    1, 0.08 
#> max values  : 15.3, 12.6 

offsets <- plot(las)
print(offsets)
#> [1]   391867.8 3901019.3

#get  the locations of the field data, need a SP data frame
# extract the coordinates of the trees and apply the shift to display the lines  in the rendering coordinate system
x <- coordinates(plotdata)[,1] - offsets[1]
y <- coordinates(plotdata)[,2] - offsets[2]
z <- plotdata$Z

# Build a GL_LINES matrix for fast rendering
x <- rep(x, each = 2)
y <- rep(y, each = 2)
tmp <- numeric(2*length(z)) 
tmp[2*1:length(z)] <- z
z <- tmp
M <- cbind(x,y,z)

# plot the point cloud with the filed data in 3d for the complete scene
offsets <- plot(las, bg = "white", size = 16)
add_treetops3d(offsets, plotdata)
rgl::segments3d(M, col = "black", lwd = 2)
````

So this all works great.

BUT, I would like the symbols of the added treetops to be smaller and possibly see trough,
Is there a way, I can change the symbols in the code above? Maybe in the add_treetops3d() section?

Thanks for your help.
Best wishes,
Linda

**Comment by Jean-Romain** [comment]
`...` is passed to `rgl:: spheres3d` You can control e.g. `radius`, `alpha` and much more.

```
add_treetops3d(x, ttops, radius = 1, fastTransparency = TRUE, alpha = 0.5)
```
